### PR TITLE
Kotlin 1.6.21 & HMPP

### PR DESCRIPTION
**Pull request by martinbonnin**
- use Kotlin 1.6.20
- make ios-test pass
- better workaround

See also https://github.com/apollographql/apollo-kotlin/issues/4019

### CODE REVIEW

File: build-logic/src/main/kotlin/TargetCompatibility.kt [hunk 1]
```diff
@@ -10,13 +10,7 @@ fun Project.configureJavaAndKotlinCompilers() {
     // For Kotlin JVM projects
     tasks.withType(KotlinCompile::class.java) {
       it.kotlinOptions {
-        freeCompilerArgs = freeCompilerArgs +
-            "-Xopt-in=kotlin.RequiresOptIn" +
-            // This is a workaround for https://youtrack.jetbrains.com/issue/KT-47000 (fixed in Koltin 1.6.20)
-            // Since we don't use @JvmDefault anywhere, the option has no effect, but suppresses the bogus compiler error
-            // See also https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-default/
-            // See also https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/
-            "-Xjvm-default=compatibility"
```
Comment (martinbonnin) [inline build-logic/src/main/kotlin/TargetCompatibility.kt:19]: @BoD is it ok to remove this?

**Comment by martinbonnin** [comment]
Ouch. This seems to break the IDE a bunch :/. We might want to wait a bit before we enable this...

Screenshot 2022-04-14 at 11 52 57

File: build-logic/src/main/kotlin/TargetCompatibility.kt (hunk 1 above)
Comment (BoD) [reply build-logic/src/main/kotlin/TargetCompatibility.kt:19]: Yes, should be good 👍

File: tests/build.gradle.kts [hunk 2]
```diff
@@ -28,6 +28,21 @@ subprojects {
     }
   }
 
+  // Workaround for https://youtrack.jetbrains.com/issue/KT-51970
```
Comment (BoD) [inline tests/build.gradle.kts:31]: _Oh my..._ 😅

**Review by BoD — APPROVED** [review]

**Comment by martinbonnin** [comment]
Just added okio 3.1.0 and HMPP support but somehow IDE is still not working. I opened this as a follow up https://youtrack.jetbrains.com/issue/KT-52059/

File: gradle.properties [hunk 3]
```diff
@@ -21,5 +21,4 @@ org.gradle.jvmargs=-Xmx8g
 # silence the "multiplatform alpha" warning
 kotlin.mpp.stability.nowarn=true
 
-# https://github.com/gradle/gradle/issues/11412
-systemProp.org.gradle.internal.publish.checksums.insecure=true
```
Comment (martinbonnin) [inline gradle.properties:25]: Another workaround bites the dust. This shouldn't be needed anymore now that we're publishing to MavenCentral only

**Comment by martinbonnin** [comment]
On hold for https://youtrack.jetbrains.com/issue/KTIJ-21583

**Comment by martinbonnin** [comment]
Marked as ready for review again as we don't need to wait for https://youtrack.jetbrains.com/issue/KTIJ-21583. We can actually build with HMPP-ready `1.6.20` and develop in IDEA with `1.6.10` based on the `idea.sync.active` property. It's not pretty but it works!

**Review by BoD — APPROVED** [review]
Hope we can remove the workarounds soon 🤞 but it's great that it works! 🎉

**Comment by martinbonnin** [comment]
> Hope we can remove the workarounds soon 🤞 but it's great that it works! 🎉
🤞

**Pull request merged**